tools/cli/main.js

```diff
@@ -768,33 +768,40 @@ Fiber(function () {
     rawArgs.push(term);
   }
 
-  // Prevent running meteor as root
-  if (process.getuid) {
-    // On UNIX platforms.
-    if (process.getuid() === 0){
-      if (_.has(rawOptions, '--unsafe-perm')) {
-        // Meteor is running as root and has the --unsafe-perm flag, just notice.
-        Console.info("");
-        Console.info("You have run Meteor as root.",
-        "Your permissions in your app directory will be incorrect if you ever attempt to perform any Meteor tasks as your non-root user.",
-        "You probably didn't want this, but you can fix it by running the following from the root of your project:");
-        Console.info("");
-        Console.info("sudo chown -Rh <username> .meteor/local");
-        Console.info("");
-        // Fix issue when running `sudo meteor --unsafe-perm`, --unsafe-perm: unknown option.
-        delete rawOptions['--unsafe-perm'];
-      }
-      else {
-        // Meteor is running as root without the --unsafe-perm flag, notice and stop.
-        Console.error("");
-        Console.error("You are attempting to run Meteor as the \"root\" user.",
+  if (_.has(rawOptions, "--unsafe-perm")) {
+    process.env.METEOR_UNSAFE_PERM = "true";
+    delete rawOptions["--unsafe-perm"];
+  }
+
+  // Prevent running meteor as root on UNIX platforms.
+  if (process.getuid &&
+      process.getuid() === 0) {
+    const unsafePerm = !! (
+      process.env.METEOR_UNSAFE_PERM &&
+      JSON.parse(process.env.METEOR_UNSAFE_PERM));
+
+    if (! unsafePerm) {
+      // Meteor is running as root without $METEOR_UNSAFE_PERM, notice and stop.
+      Console.error("");
+      Console.error(
+        "You are attempting to run Meteor as the root user.",
         "If you are developing, this is almost certainly *not* what you want to do and will likely result in incorrect file permissions.",
-        "However, if you are running this in a build process (CI, etc.) or you are absolutely sure you know what you are doing,",
-        "add the `--unsafe-perm` flag to this command to proceed.");
-        Console.error("");
+        "However, if you are running this command in a build process (CI, etc.), or you are absolutely sure you know what you are doing,",
+        "set the $METEOR_UNSAFE_PERM environment variable to proceed."
```

Comment:
My recommendation would be the same as @mitar on both counts.

I understand it's semantics and colloquial used to indicate "a variable" but in this case the `$` is not part of the variable in this case – even more so than say, PHP or Perl.  It's shell expansion syntax and isn't used for _setting_ (or even _getting_ in most languages).  However, this form of usage is used in several other places in the code (and `History.md`) – but it is more commonly used without.